src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+ *     possible values: start, center, end, justify. This is only necessary if you do
+ *     not want the panel to intelligently determine location based on screen edges.
+ *       - `start`:
+ *         The panel is aligned to the start of the originating element in the
+ *         x-direction. This value is the left of the element if the direction is
+ *         left-to-right and the right of the element if the direction is
+ *         right-to-left.
+ *       - `end`:
+ *         The panel is aligned to the end of the originating element in the
+ *         x-direction. This value is the right of the element if the direction is
+ *         left-to-right and the left of the element if the direction is
+ *         right-to-left.
+ *   - `alignY` - `{string=}`: What point of origin to use on the y-axis. There are 4
+ *     possible values: start, center, end, justify. This is only necessary if you do
+ *     not want the panel to intelligently determine location based on screen edges.
+ *       - `top`:
+ *         The panel is aligned to the top of the originating element.
+ *       - `bottom`:
+ *         The panel is aligned to the bottom of the originating element.
+ *   - `overlap` - `{boolean=}`: Whether the panel should overlap the originating
+ *     element. This value is true by default.
+ *   - `center` - `{boolean=}`: If true, centers the panel in the viewport, both
+ *     horizontally and vertically. If this option is set, `alignX` and `alignY` will
+ *     be ignored.
+ *   - `animation` - `{string=}`: The classname for the type of animation. There are
+ *     several built-in animations that you can use:
+ *       `md-animate-fly`: The panel flies in and out from the specified elements.
+ *       `md-animate-scale`: The panels scales in and out.
+ *       `md-animate-none`: No animations.
+ *   - `targetEvent` - `{DOMClickEvent=}`: A click's event object. When passed in as
+ *     an option, the location of the click will be used as the starting point for the
+ *     opening animation of the panel.
+ *   - `openFrom` - `{(string|Element|Object)=}`: The query selector, DOM element,
+ *     or the Rect object that is used to determine the bounds (top, left, height,
+ *     width) from which the panel will originate.
+ *   - `closeTo` - `{(string|Element|Object)=}`: The query selector, DOM element, or
+ *     the Rect object that is used to determine the bounds (top, left, height, width)
+ *     from which the panel will close to.
+ *   - `hasBackdrop` - `{boolean=}`: Whether there should be an opaque backdrop behind
+ *     the panel. Defaults to false.
+ *   - `escapeToClose` - `{boolean=}`: Whether the user can press escape to close the
+ *     panel. Defaults to false.
+ *   - `clickOutsideToClose` - `{boolean=}`: Whether the user can click outside the
+ *     panel to close it. Defaults to false.
+ *   - `disableParentScroll` - `{boolean=}`: Whether the user can scroll the page
```

Comment:
@jelbourn I'm not sure if I understand the change you are suggesting here, if any. This would be making the user be explicit about whether they want scrolling or not.